main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,33 +57,117 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        // equal distance and different coordinates
+        if (!areCloseWithDifferentCoordinates(curr, prev)) {
           noticeContainer.addValidationNotice(
-              new DecreasingOrEqualShapeDistanceNotice(
-                  curr.shapeId(),
-                  curr.csvRowNumber(),
-                  curr.shapeDistTraveled(),
-                  curr.shapePtSequence(),
-                  prev.csvRowNumber(),
-                  prev.shapeDistTraveled(),
-                  prev.shapePtSequence()));
+              new EqualShapeDistanceDiffCoordinatesNotice(prev, curr));
+        } else if (curr.shapePtLat() == prev.shapePtLat()
+            && curr.shapePtLon() == prev.shapePtLon()) {
+          // equal distance and same coordinates
+          noticeContainer.addValidationNotice(
+              new EqualShapeDistanceSameCoordinatesNotice(prev, curr));
         }
       }
     }
   }
 
   /**
-   * When sorted on `shapes.shape_pt_sequence` key, shape points should have strictly increasing
+   * Checks if two {@code GtfsShape} have different coordinates and are farther than {@code
+   * ShapeIncreasingDistanceValidator#MAX_DISTANCE_SHAPEPOINTS}.
+   *
+   * @param shape the first {@code GtfsShape}
+   * @param otherShape the other {@code GtfsShape}
+   * @return true the two {@code GtfsShape} have distinct coordinates and if the distance that
+   *     separates them is less or equal than {@code
+   *     ShapeIncreasingDistanceValidator#MAX_DISTANCE_SHAPEPOINTS}; false otherwise.
+   */
+  private static boolean areCloseWithDifferentCoordinates(GtfsShape shape, GtfsShape otherShape) {
+    return shape.shapePtLon() != otherShape.shapePtLon()
+        && shape.shapePtLat() != otherShape.shapePtLat()
```

Comment:
You can have different coordinates while either lon or lat is identical. Shouldn't this be `!(shape.shapePtLon() == otherShape.shapePtLon() && shape.shapePtLat() == otherShape.shapePtLat())`

Reply:
But it looks like this function isn't used anywhere